lib/iris/experimental/ugrid.py

```diff
@@ -1525,45 +1516,53 @@ def filters(
         edge=None,
         face=None,
     ):
+        # TBD: support coord_systems?
+
         # rationalise the tri-state behaviour
         args = [node, edge, face]
         state = not any(set(filter(lambda arg: arg is not None, args)))
         node, edge, face = map(
             lambda arg: arg if arg is not None else state, args
         )
 
-        def func(args):
-            return args[1] is not None
+        def populated_coords(coords_tuple):
+            return list(filter(None, list(coords_tuple)))
 
-        members = {}
+        members = []
         if node:
-            members.update(
-                dict(filter(func, self.node_coords._asdict().items()))
-            )
+            members += populated_coords(self.node_coords)
         if edge:
-            members.update(
-                dict(filter(func, self.edge_coords._asdict().items()))
-            )
+            members += populated_coords(self.edge_coords)
         if hasattr(self, "face_coords"):
             if face:
-                members.update(
-                    dict(filter(func, self.face_coords._asdict().items()))
-                )
+                members += populated_coords(self.face_coords)
         else:
             dmsg = "Ignoring request to filter non-existent 'face_coords'"
             logger.debug(dmsg, extra=dict(cls=self.__class__.__name__))
 
-        result = self._filters(
+        if axis is not None:
+            axis = axis.upper()
+            members = [
+                instance_
+                for instance_ in members
+                if guess_coord_axis(instance_) == axis
+            ]
```

Comment:
I originally left it out because it's not relevant to the `Connectivity` class, but since it is relevant to the two other uses in `Cube` and `Mesh` coord managers, maybe I should include `axis` and just pass `None` for this parameter where it isn't relevant?

Reply:
6675cd8a52c69f5b6f5cea55d93a86392c6ffd3d